src/ampycloud/data.py

```diff
@@ -93,16 +94,33 @@ def _cleanup_pdf(self, data: pd.DataFrame) -> pd.DataFrame:
         # Begin with a thorough inspection of the dataset
         data = utils.check_data_consistency(data, req_cols=self.DATA_COLS)
 
+        # By default we set NSC to false, NOTE: attrs is still experimental in pandas!
+        try:
+            data.attrs['high_clouds_detected'] = False
+        except AttributeError:
+            # Catch for older versions of pandas that do not support DataFrame.attrs
+            # NOTE: Untestable as we cannot monkeypatch attrs as it would result
+            # in side-effects for __repr__ which break other tests
+            with warnings.catch_warnings():
+                # pandas assumes we are trying to assign a column, which is not the case
+                warnings.simplefilter("ignore")
+                data.attrs = {'high_clouds_detected': False}
+
         # Then also drop any hits that is too high
         if self.msa is not None:
             hit_alt_lim = self.msa + self.msa_hit_buffer
             logger.info('Cropping hits above MSA+buffer: %s ft', str(hit_alt_lim))
             # Type 1 or less hits above the cut threshold get turned to NaNs, to signal a
             # non-detection below the MSA. Also change the hit type to 0 accordingly !
-            data.loc[data[(data.alt > hit_alt_lim) & (data.type <= 1)].index, 'type'] = 0
-            data.loc[data[(data.alt > hit_alt_lim) & (data.type <= 1)].index, 'alt'] = np.nan
+            above_msa_t1_or_less = data[(data.alt > hit_alt_lim) & (data.type <= 1)].index
```

Comment:
I am confused about this part: data.type <= 1 includes -1 (VV hit), 0 (no hit) and 1 (first cloud base). I do not understand why we separate data this way (type [-1, 0, 1] at line 115 vs type [2, 3] at line 119)

It was already like this in the previous version, but I am wondering if we are doing things correctly. Using -1 for VV might be more error prone when using "lower than" conditions. Use 4 for VV hit instead? (tbd with @fpavogt)

Reply:
The motivation for using -1 for VV hits is that some ceilometers may possibly return more than 3 simultaneous base heights per time step (either now, or in the future). I find using -1 much safer from this perspective, because it is clearly "unique".

The separation at line 115 comes from the fact that we cannot simply "remove" the cloud base detections above the MSA from the dataset: we need to replace them with a non-detection (else, we would over-estimate the density of the other layers). To replace a cloud detection with a non-detection, we must set its type to 0 - but only once per time step. For cases with multiple base detections (with type 1, 2, 3), a single entry must be turned to type 0, and all the others must be cropped from the dataset (else, we would have created duplicated hits of type 0).

Reply:
Nothing to add - I agree with the reasoning as to why set -1 for VV hits. I'll add a comment on why we are changing types.